ios/testmanagerd/xctestrunutils.go

```diff
@@ -0,0 +1,173 @@
+package testmanagerd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/danielpaulus/go-ios/ios"
+	"howett.net/plist"
+	"io"
+	"maps"
+	"os"
+	"path/filepath"
+)
+
+// XCTestRunData represents the structure of an .xctestrun file
+type XCTestRunData struct {
+	TestConfig        SchemeData        `plist:"-"`
+	XCTestRunMetadata XCTestRunMetadata `plist:"__xctestrun_metadata__"`
+}
+
+func (data XCTestRunData) toTestConfig(device ios.DeviceEntry, listener *TestListener) (TestConfig, error) {
```

Comment:
I think we should move this method down, for the sake of readability.

Reply:
moved this down and renamed the function